featurebyte/routes/target_table/api.py

```diff
@@ -29,136 +30,182 @@
+        sort_by: Optional[str] = SortByQuery,
+        sort_dir: Optional[str] = SortDirQuery,
+        search: Optional[str] = SearchQuery,
+        name: Optional[str] = NameQuery,
+    ) -> TargetTableList:
+        """
+        List TargetTables
+        """
+        controller = request.state.app_container.target_table_controller
+        target_table_list: TargetTableList = await controller.list(
+            page=page,
+            page_size=page_size,
+            sort_by=sort_by,
+            sort_dir=sort_dir,
+            search=search,
+            name=name,
+        )
+        return target_table_list
+
+    @staticmethod
+    async def list_target_table_audit_logs(
+        request: Request,
+        target_table_id: PydanticObjectId,
+        page: int = PageQuery,
+        page_size: int = PageSizeQuery,
+        sort_by: Optional[str] = AuditLogSortByQuery,
+        sort_dir: Optional[str] = SortDirQuery,
+        search: Optional[str] = SearchQuery,
+    ) -> AuditDocumentList:
+        """
+        List TargetTable audit logs
+        """
+        controller = request.state.app_container.target_table_controller
+        audit_doc_list: AuditDocumentList = await controller.list_audit(
+            document_id=target_table_id,
+            page=page,
+            page_size=page_size,
+            sort_by=sort_by,
+            sort_dir=sort_dir,
+            search=search,
+        )
+        return audit_doc_list
+
+    @staticmethod
+    async def get_target_table_info(
+        request: Request, target_table_id: PydanticObjectId, verbose: bool = VerboseQuery
+    ) -> TargetTableInfo:
+        """
+        Get TargetTable info
+        """
+        controller = request.state.app_container.target_table_controller
+        info = await controller.get_info(document_id=target_table_id, verbose=verbose)
+        return cast(TargetTableInfo, info)
+
+    @staticmethod
+    async def download_table_as_pyarrow_table(
+        request: Request, target_table_id: PydanticObjectId
+    ) -> StreamingResponse:
+        """
+        Download TargetTable as pyarrow table
+        """
+        controller = request.state.app_container.target_table_controller
+        result: StreamingResponse = await controller.download_materialized_table(
+            document_id=target_table_id,
+            get_credential=request.state.get_credential,
+        )
+        return result
+
+    @staticmethod
+    async def update_target_table_description(
+        request: Request,
+        target_table_id: PydanticObjectId,
+        data: DescriptionUpdate,
+    ) -> TargetTableModel:
+        """
+        Update target_table description
+        """
+        controller = request.state.app_container.target_table_controller
+        target_table: TargetTableModel = await controller.update_description(
+            document_id=target_table_id,
+            description=data.description,
+        )
+        return target_table
```

Comment:
If i understood this correctly, the main purpose of doing this is to simplify the flow to overwrite controller logic at the `featurebyte-app` side, right? I think current flow make sense.

Reply:
oops finally getting back to this PR - but yea, overwriting should be easier (more object oriented), and it should also be easier to spin up the basic CRUD endpoints that are pretty standardized across our collections/fastapi endpoints.